moniker_src/data_moniker_service.cpp

```diff
@@ -0,0 +1,294 @@
+
+        std::cout << "Starting thread with RunSidebandReadWriteLoop()" << std::endl;
+        auto thread = std::make_shared<std::thread>(&DataMonikerService::RunSidebandReadWriteLoop, this, identifier, strategy, readers, writers);
+        thread->detach();
+
+        std::cout << "Exit BeginSidebandStream" << std::endl;
+        return Status::OK;
+    }
+
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
+    Status DataMonikerService::StreamReadWrite(ServerContext* context, ServerReaderWriter<MonikerReadResponse, MonikerWriteRequest>* stream)
+    {
+        EndpointList writers;
+        EndpointList readers;
+        MonikerWriteRequest writeRequest;
+        stream->Read(&writeRequest);
+        InitializeMonikerList(writeRequest.monikers(), &readers, &writers);
+
+        google::protobuf::Arena arena;
+        while (stream->Read(&writeRequest) && !context->IsCancelled())
+        {
+            int x = 0;
+            for (auto writer: writers)
+            {
+                std::get<0>(writer)(std::get<1>(writer), arena, const_cast<google::protobuf::Any&>(writeRequest.data().values(x++)));
+            }
+
+            MonikerReadResponse readResult;
+            for (auto reader: readers)
+            {
+                auto readValue = readResult.mutable_data()->add_values();
+                std::get<0>(reader)(std::get<1>(reader), arena, *readValue);
+            }
+            stream->Write(readResult);
+            arena.Reset();
+        }
+        return Status::OK;
+    }
+
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
+    Status DataMonikerService::StreamRead(ServerContext* context, const MonikerList* request, ServerWriter<MonikerReadResponse>* writer)
+    {
+        EndpointList writers;
+        EndpointList readers;
+        InitializeMonikerList(*request, &readers, &writers);
+
+        google::protobuf::Arena arena;
+        while (!context->IsCancelled())
+        {
+            MonikerReadResponse readResult;
+            for (auto reader: readers)
+            {
+                auto readValue = readResult.mutable_data()->add_values();
+                std::get<0>(reader)(std::get<1>(reader), arena, *readValue);
+            }
+            writer->Write(readResult);
+            arena.Reset();
+        }
+        return Status::OK;
+    }
+
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
+    Status DataMonikerService::StreamWrite(ServerContext* context, ServerReaderWriter<StreamWriteResponse, MonikerWriteRequest>* stream)
+    {
+    #ifndef _WIN32
+        cpu_set_t cpuSet;
+        CPU_ZERO(&cpuSet);
+        CPU_SET(1, &cpuSet);
+        sched_setaffinity(0, sizeof(cpu_set_t), &cpuSet);
```

Comment:
Same comment as above.  We can't really make this official.  We need to understand what the perf difference is and if we need it.  If we do need it we should figure out what processor to use.